request.go

```diff
@@ -229,6 +241,8 @@ func (r *Request) close() error {
 		}
 	}()
 
+	r.state.closeListerAt()
```

Comment:
```
err := r.state.closeListerAt()
```

Perhaps? This obviates the need for line 248 as well.

Reply:
maybe we can just check for io.Closer like for readers/writers instead of extending the interface

Reply:
Yeah, I think that’s a simpler mor direct change.

Reply:
> Yeah, I think that’s a simpler mor direct change.

Done